test/test_integration.py

```diff
@@ -1021,6 +1005,10 @@ def display_data(n_intervals):
         # get the output after it's stopped, it shouldn't be higher than before
         output_stopped = self.wait_for_element_by_css_selector('#output')
 
+        self.wait_for_text_to_equal("#output", output_stopped.text)
+
+        # This test logic is bad
+        # same element check for same text will always be true.
         self.assertEqual(output.text, output_stopped.text)
```

Comment:
Can we not remove line `1012` then, since you added `wait_for_text_to_equal` on line `1008`?

Reply:
The wait_for_text I added is as bad, it check the text against itself. I think we should refactor the last few lines of that test.

Reply:
Ok sounds good!

Reply:
💃